File: go/isogram/isogram.go
// Package isogram implements isogram checking
package isogram

import (
	"regexp"
	"strings"
)

// IsIsogram checks if a given phrase is isogram
func IsIsogram(phrase string) bool {
	r := regexp.MustCompile("[a-z]")
	s := strings.ToLower(phrase)
	alphabets := strings.Join(r.FindAllString(s, -1), "")

	for _, c := range s {
		if strings.Count(alphabets, string(c)) > 1 {
			return false
		}
	}
	return true
}

File: kotlin/isogram/src/main/kotlin/Isogram.kt
object Isogram {
    fun isIsogram(word: String): Boolean {
        val filtered = word.filterAlphabet()
        return filtered.all { c -> filtered.count { it == c } <= 1 }
    }
}

private fun String.filterAlphabet() = this.toLowerCase().replace("[^a-z]".toRegex(), "")

